### DISCUSSION TITLE
Couchdb $in max elements limitation

### DISCUSSION THREAD

**Discussion by volodyad** [apache/couchdb#3030] [Q&A]
Hello,

Is there any limitation around how many elements could be inside $in query?

**Comment by janl** [comment]
There is no defined limit, but the implementation is such that for each row in your index, all elements of your $in set will be applied in turn. So execution speed grows O(N*M) (N = $in set length, M = index length. And sending megabytes or more of a JSON $in array might lead to request timeouts and CPU spikes elsewhere in your system.

How many items to you wanna send?
